Fail if given Schedule as input circuit to IBMBackend.run()

merav-aharoni opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary
Raise exception if given Schedule as input circuit to `IBMBackend.run()`. Refer the user to the tutorial on using pulse gates.


### Details and comments
See issue #602 and also #593.

kt474 reviewed (commented):
There are a couple more places that mention `schedule` that should be removed -
[L82](https://github.com/Qiskit/qiskit-ibm-provider/blob/34e4f622a3052df9f7ac93c7928e3fda02af0583/qiskit_ibm_provider/ibm_backend.py#L82), [L101](https://github.com/Qiskit/qiskit-ibm-provider/blob/34e4f622a3052df9f7ac93c7928e3fda02af0583/qiskit_ibm_provider/ibm_backend.py#L101),  [L332](https://github.com/Qiskit/qiskit-ibm-provider/blob/34e4f622a3052df9f7ac93c7928e3fda02af0583/qiskit_ibm_provider/ibm_backend.py#L332)

merav-aharoni commented:
Thanks for pointing those out.
I will fix those, and also noticed some other defects, so will fix and will let you know when done.

merav-aharoni commented:
@kt474 - I fixed two of the lines you pointed out. Regarding https://github.com/Qiskit/qiskit-ibm-provider/blob/04e534b94f0bd686341d800d3df72cab0e893c90/qiskit_ibm_provider/ibm_backend.py#L331 -  I left this intentionally, so that I could raise a relevant error.
I am not sure if what we want here is an error or should this be a deprecation message? I did an error, because that is what was specified in the issue.
I also added a test, although this test can probably be removed at some point.

jyu00 commented on qiskit_ibm_provider/ibm_backend.py:
```diff
@@ -482,11 +485,23 @@ def run(
-        if isinstance(circuits, (QuantumCircuit, Schedule)):
+        if isinstance(circuits, Schedule):
+            raise IBMBackendValueError(
+                "Class 'Schedule' is no longer supported as "
+                "an input circuit. See 'run' method documentation"
```
Why not mention the pulse gate with link right in the message? So users don't need to hunt down the docs

jyu00 commented on releasenotes/notes/remove_schedule_from_backend_run-6b3c771799db9e9a.yaml:
```diff
@@ -0,0 +1,6 @@
+---
+features:
```
I don't think this is a feature lol. I'm pretty sure there's a "removal" section.

merav-aharoni replied:
I couldn't find a `removal` section. Changed to `deprecations`.

jyu00 commented:
> I am not sure if what we want here is an error or should this be a deprecation message? I did an error, because that is what was specified in the issue.

Yeah unfortunately deprecation is not good enough, since we use qpy to serialize the circuits sent to the server, and qpy doesn't support `Schedule`.
